packages/design-system-react/src/components/icon/template.d.ts

```diff
@@ -0,0 +1,20 @@
+type TemplateVariables = {
+  imports: string;
+  interfaces: string;
+  componentName: string;
+  props: string;
+  jsx: string;
+  exports: string;
+};
+
+type TemplateContext = {
+  tpl: (strings: TemplateStringsArray, ...values: unknown[]) => string;
+  options?: Record<string, unknown>;
+};
+
+declare const template: (
+  variables: TemplateVariables,
+  context: TemplateContext,
+) => string;
+
+export default template;
```

Comment:
The template.d.ts file is needed alongside template.js because:
1. SVGR requires the template file to be JavaScript (.js), as it's used in their build pipeline
2. However, we still want TypeScript type checking and IDE support when working with the template
3. The .d.ts declaration file provides type definitions for:
- Template variables (imports, interfaces, component name, etc.)
- Template context (tpl function and options)
This setup gives us the best of both worlds: compatibility with SVGR's requirements while maintaining type safety in our codebase.